File: Ejercicios/ejerciciosRuby/pkg/civitas-0.0.1/lib/parcela.rb
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.
# encoding:utf-8
module EjerciciosRuby
class Parcela
  
    attr_reader :nombre, :precioCompra, :precioEdificar, :precioAlquilerCompleto, :numCasas, :numHoteles  #atributos de instancia en java
    
    
    def initialize( nombre, precioCompra, precioEdificar, precioBaseAlquiler)   #una clase con los atributos previos y le añadimos un constructor que reciba  todo eso
        
       @nombre = nombre
       @precioCompra = precioCompra
       @precioEdificar = precioEdificar
       @precioBaseAlquiler = precioBaseAlquiler
       @@FACTORALQUILERCALLE = 1.0
       @@FACTORALQUILERCASA = 1.0;
       @@FACTORALQUILERHOTEL = 4.0;
       @num_casa = @num_hoteles = 0 
    end
    
    
    def precioAlquilerCompleto
      @precioBaseAlquiler*(@@FACTORALQUILERCALLE;+@numCasas +@numHoteles*@@FACTORALQUILERHOTEL;)
     
      
     
    end
    
    def construirHotel ()
      @numHoteles =+ 1
      
      return true
    end
    
    def construirCasa ()
      @numCasas =+ 1
      
      return true
    end
    
   def to_s
     salida= "Nombre de la parcela: #{@nombre}
     Precio de la parcela: #{@precioCompra}
     Alquiler de la parcela: #{self.precioAlquilerCompleto()}
     Edificar en la parcela: #{@precioEdificar}
     Casas de la parcela: #{@numCasas}
     Hoteles de la parcela: #{@numHoteles}"
   end
  end
end

File: Ejercicios/ejerciciosJava/src/ejerciciosS1/Parcela.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */

package ejerciciosS1;

/**
 *
 * @author Luis
 */
public class Parcela {
    private String nombre;
    private float precioCompra;
    private float precioEdificar;  ///TODO ESYO SON ATRIBUTOS DE INSTANCIA
    private float precioBaseAlquiler;
    private int numCasas;
    private int numHoteles;
    static private float FACTORALQUILERCALLE = 1.0f;
    static private float FACTORALQUILERCASA = 1.0f;
    static private float FACTORALQUILERHOTEL = 4.0f;
    
    
    
    //una clase con los atributos previos y le añadimos un constructor que reciba  todo eso
   
    Parcela (String _nombre, float _precioCompra, float _precioEdificar, float _precioBaseAlquiler ) {
        
        
        this.nombre = _nombre;
        this.precioCompra = _precioCompra;
        this.precioEdificar = _precioEdificar;
        this.precioBaseAlquiler = _precioBaseAlquiler;
        this.numCasas = 0;
        this.numHoteles = 0;
       
        
    }
    
     String getNombre () { return nombre;   //CONSULTORES
     }
     
      float getPrecioCompra () {
         return precioCompra;
     }
     
      float getPrecioEdificar () {
         return precioEdificar;
         
     }
     
      int getNumCasas () {
         return numCasas;
        
     }
      int getNumHoteles () {
          return numHoteles;
          
      }
     
     
   public  float getPrecioAlquilerCompleto () {    //calcula y devuelve el precio completo de alquilar dicha parcela teniendo en cuenta tanto
       //el solar como sus edificaciones.
       
       
       return (precioBaseAlquiler*FACTORALQUILERCALLE*(1+numCasas*FACTORALQUILERCASA + (numHoteles*FACTORALQUILERHOTEL*4)));
       
}
   
   boolean construirCasa (){
       
      numCasas++;
      
       return true;
        
   }
   
   boolean construirHotel (){
       numHoteles++;
       return true;
   }
   
   boolean igualdadIdentidad(Parcela otraParcela){
      boolean resultado = (this == otraParcela)?true:false; //devuelve a nivel de identidad(comparandolo con this) si el parametro es igual al receptor del mensaje
      //esta parcela
      return resultado;
   }
   public boolean igualdadEstado (Parcela otraParcela){
       boolean resultado = (this.equals(otraParcela))?true:false; //devuelve truie si el objeto receptor del mensaje es igual a cuanto estado, lo mismo pero con equals
       return resultado;
   }
   

}